### PR TITLE
mattermost-desktop: 4.6.2 -> 5.0.3

### PR DESCRIPTION
**Pull request by liff**
###### Motivation for this change

New [major release of the desktop app](https://docs.mattermost.com/install/desktop-app-changelog.html).

While at it I formatted the source with `nixpkgs-fmt` and changed to use `atomEnv` since it’s an Electron app.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by iAmMrinal0** [comment]
Result of `nixpkgs-review pr 144266` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mattermost-desktop</li>
  </ul>
</details>

**Comment by iAmMrinal0** [comment]
GUI works and has the version as mentioned in the PR.

File: pkgs/applications/networking/instant-messengers/mattermost-desktop/default.nix [hunk 1]
```diff
@@ -1,105 +1,90 @@
...
-        $out/share/mattermost-desktop/mattermost-desktop
+    patchShebangs $out/share/mattermost-desktop/create_desktop_file.sh
+    $out/share/mattermost-desktop/create_desktop_file.sh
+    rm $out/share/mattermost-desktop/create_desktop_file.sh
+    mkdir -p $out/share/applications
+    chmod -x Mattermost.desktop
+    mv Mattermost.desktop $out/share/applications/Mattermost.desktop
+    substituteInPlace \
+      $out/share/applications/Mattermost.desktop \
+      --replace /share/mattermost-desktop/mattermost-desktop /bin/mattermost-desktop
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/mattermost-desktop/default.nix:78]:
```suggestion
    substituteInPlace $out/share/applications/Mattermost.desktop \
      --replace /share/mattermost-desktop/mattermost-desktop /bin/mattermost-desktop
```

File: pkgs/applications/networking/instant-messengers/mattermost-desktop/default.nix [hunk 2]
```diff
@@ -1,105 +1,90 @@
...
-      $out/share/mattermost-desktop/create_desktop_file.sh
-      rm $out/share/mattermost-desktop/create_desktop_file.sh
-      mkdir -p $out/share/applications
-      mv Mattermost.desktop $out/share/applications/Mattermost.desktop
-      substituteInPlace \
-        $out/share/applications/Mattermost.desktop \
-        --replace /share/mattermost-desktop/mattermost-desktop /bin/mattermost-desktop
+    mkdir -p "$out/bin"
+    ln -s $out/share/mattermost-desktop/mattermost-desktop \
+      $out/bin/mattermost-desktop
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/mattermost-desktop/default.nix:68]:
```suggestion
    ln -s $out/share/mattermost-desktop/mattermost-desktop $out/bin/mattermost-desktop
```

File: pkgs/applications/networking/instant-messengers/mattermost-desktop/default.nix [hunk 3]
```diff
@@ -1,105 +1,90 @@
...
+      fetchurl
+        {
           url = "https://releases.mattermost.com/desktop/${version}/${pname}-${version}-linux-ia32.tar.gz";
-          sha256 = "04jv9hkmkh0jipv0fjdprnp5kmkjvf3c0fah6ysi21wmnmp5ab3m";
+          sha256 = "0y48yd0qxkcz3vv62m9qg2qmji563nbpb17lragk1ykxvj32373g";
         }
-      else
-        throw "Mattermost-Desktop is not currently supported on ${stdenv.hostPlatform.system}";
+    else
+      throw "Mattermost-Desktop is not currently supported on ${stdenv.hostPlatform.system}";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/mattermost-desktop/default.nix:35]: Maybe we could use a set instead of an if else chain.

Comment (liff) [reply pkgs/applications/networking/instant-messengers/mattermost-desktop/default.nix]: Changed to set, thanks.

**Comment by happyalu** [comment]
Found this PR when looking for the latest version :) Is this blocked on something?

**Comment by liff** [comment]
Just busy reviewers, I imagine. Bumped to 5.0.2 in the meantime :)

**Comment by happyalu** [comment]
I tested a built from this PR and 5.0.2 is working well for me on x86_64-linux.

@SuperSandro2000 (ping. sorry!)

File: pkgs/applications/networking/instant-messengers/mattermost-desktop/default.nix [hunk 4]
```diff
@@ -1,105 +1,89 @@
...
+      sha256 = "1a91b2sqkz3kr324qy26325sfjgjr8k07i5qh54vgq1cxbig2mgf";
+    };
+  };
+
+in
+
+stdenv.mkDerivation {
+  inherit pname version;
+
+  src = fetchurl srcs.${stdenv.hostPlatform.system};
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/mattermost-desktop/default.nix:36]: Please add an or throw unsupported system that evals on other platforms do not end in an syntax error.

Comment (liff) [reply pkgs/applications/networking/instant-messengers/mattermost-desktop/default.nix]: Added.

**Comment by miallo** [comment]
Working nicely :) Thanks!

**Comment by happyalu** [comment]
@jokogr @SuperSandro2000 could you please take a look at this again? Thanks!

File: pkgs/applications/networking/instant-messengers/mattermost-desktop/default.nix [hunk 5]
```diff
@@ -1,105 +1,91 @@
...
+  };
+
+  inherit (stdenv.hostPlatform) system;
+
+in
+
+stdenv.mkDerivation {
+  inherit pname version;
+
+  src = fetchurl (srcs."${stdenv.hostPlatform.system}" or (throw "Unsupported system ${system}"));
```
Comment (miallo) [inline pkgs/applications/networking/instant-messengers/mattermost-desktop/default.nix:38]: I am not too experienced with nix, so forgive me if this is a stupid question: Couldn't `${stdenv.hostPlatform.system}` be replaced with `${system}` (just for readability)?

**Review by miallo — COMMENTED** [review]
It looks reasonable to me (and it is working), but I am not very familiar with nix, so I might have overlooked something

**Comment by liff** [comment]
5.0.3 was just released, so bumped.

**Pull request merged**